core/lowering/passes/remove_unnecessary_casts.cpp

```diff
@@ -211,6 +212,150 @@ void RemoveSingleUse0DTensors(std::shared_ptr<torch::jit::Graph>& g) {
   LOG_GRAPH("Post removing single use 0-dim Tensor operations: " << *g);
 }
 
+// Schemas for Aten::Int which can be replaced by scalar equivalents
+const std::unordered_set<c10::Symbol> AtenIntReplacementNodeKinds = {
+    torch::jit::aten::mul,
+    torch::jit::aten::floor_divide,
+};
+
+c10::optional<torch::jit::Value*> Validate0DTensor(torch::jit::Value* value) {
```

Comment:
Refactored to use `c10::optional` wrapper instead of `nullptr` + replaced pointer checks with `.has_value()`